File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int curSum = 0;
        int maxSum = Integer.MIN_VALUE;

        for(int n : nums){
            maxSum = Math.max(maxSum,n);
        }

        for(int n : nums){
            curSum += n;
            maxSum = Math.max(maxSum,curSum);

            if(curSum < 1){
                curSum = 0;
            }
        }
        return maxSum;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let maxSum = Number.MIN_SAFE_INTEGER;
    let curSum = 0;

    for(let n of nums){
        maxSum = Math.max(maxSum,n);
    }

    for(let n of nums){
        curSum += n;
        maxSum = Math.max(curSum,maxSum);

        if(curSum < 0){
            curSum = 0;
        }
    }
    return maxSum;
};